python/kserve/kserve/model_server.py

```diff
@@ -239,6 +252,23 @@ def __init__(
                 logging.configure_logging(args.log_config_file)
         self.access_log_format = access_log_format
         self._custom_exception_handler = None
+        predictor_config = PredictorConfig(
+            predictor_host=args.predictor_host,
+            predictor_protocol=args.predictor_protocol,
+            predictor_use_ssl=args.predictor_use_ssl,
+            predictor_request_timeout_seconds=args.predictor_request_timeout_seconds,
+            predictor_request_retries=args.predictor_request_retries,
+            predictor_health_check=args.enable_predictor_health_check,
+        )
+
+        # Enable predictor health check if transformer and predictor is collocated
+        if args.predictor_host is not None and any(
+            val in args.predictor_host.lower() for val in ["localhost", "127.0.0.1"]
+        ):
+            predictor_config.predictor_health_check = True
```

Comment:
Let's do not make this as a special case, otherwise there is no way to turn off health check if there is an issue for the collocated mode